core/extrinsics_submission_service/network_address.hpp

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_EXTRINSICS_SUBMISSION_SERVICE_NETWORK_ADDRESS_HPP
+#define KAGOME_CORE_EXTRINSICS_SUBMISSION_SERVICE_NETWORK_ADDRESS_HPP
+
+#include <cstdint>
+
+namespace kagome::service {
+  enum class IpVersion { IPV4, IPV6 };
+
+  struct NetworkAddress {
+    IpVersion ip_version{IpVersion::IPV4};
+    uint16_t port{0};
```

Comment:
you can actually use `boost::asio::ip::address` instead, and support both ipv4 && ipv6

Reply:
it is okay to expose asio details in the implementation (e.g. make address as a constructor argument of implementation, in the interface use only start or listen)

Reply:
ok